### PR TITLE
LA-585 Refactor Upload Request Group

### PR DESCRIPTION
**Pull request by huynguyennovem**

### CODE REVIEW

File: lib/presentation/widget/upload_request_group/created/created_upload_request_group_widget.dart [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+  Widget _buildDefaultViewWidget() {
+    return StoreConnector<AppState, CreatedUploadRequestGroupState>(
+      converter: (store) => store.state.createdUploadRequestGroupState,
+      builder: (_, state) => Column(children: [
+				_buildMenuSorterPending(),
+				Expanded(
+						child: RefreshIndicator(
+								onRefresh: () async => _model.getUploadRequestCreatedStatus(),
+								child: _buildListData(state.uploadRequestsCreatedList)))
+			])
```
Comment (hoangdat) [inline lib/presentation/widget/upload_request_group/created/created_upload_request_group_widget.dart:99]: Should separate in other method `_buildCreatedUploadRequest`
```
Widget _buildCreatedUploadRequest() {
  return Column(...);
}

File: lib/presentation/widget/upload_request_group/created/created_upload_request_group_widget.dart [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+  }
+
+  Widget _buildListData(List<SelectableElement<UploadRequestGroup>> uploadRequestList) {
+    if (uploadRequestList.isEmpty) {
+      return _buildCreateUploadRequestsHere(context);
+    } else {
+      return ListView.builder(
+          itemCount: uploadRequestList.length, 
+					itemBuilder: (context, index) => 
+						_buildPendingUploadTile(uploadRequestList[index]));
```
Comment (hoangdat) [inline lib/presentation/widget/upload_request_group/created/created_upload_request_group_widget.dart:124]: indent

File: lib/presentation/widget/upload_request_group/created/created_upload_request_group_widget.dart [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+				if (state.uploadRequestsCreatedList.isEmpty) {
+					return _buildNoResultFound();
+				}
+				return Column(children: [
+					_buildResultCount(state.uploadRequestsCreatedList.length),
+					Expanded(
+						child: _buildListData(state.uploadRequestsCreatedList))
+				]);
+			});
+  }
```
Comment (hoangdat) [inline lib/presentation/widget/upload_request_group/created/created_upload_request_group_widget.dart:184]: indent

Comment (huynguyennovem) [reply lib/presentation/widget/upload_request_group/created/created_upload_request_group_widget.dart:184]: I don't know why but it is good indent in my local machine.

Screen Shot 2021-08-23 at 11 35 06

File: lib/presentation/widget/upload_request_group/created/created_upload_request_group_widget.dart (hunk 2 above)
Comment (huynguyennovem) [reply lib/presentation/widget/upload_request_group/created/created_upload_request_group_widget.dart:124]: In my local machine

Screen Shot 2021-08-23 at 11 36 48

File: lib/presentation/widget/upload_request_group/created/created_upload_request_group_widget.dart [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+        child: Align(
+          alignment: Alignment.centerLeft,
+          child: Text(
+            AppLocalizations.of(context).results_count(count),
+            style: TextStyle(fontSize: 16.0, color: AppColor.searchResultsCountTextColor),
+          ),
+        ),
+      ),
+    );
+  }
```
Comment (hoangdat) [inline lib/presentation/widget/upload_request_group/created/created_upload_request_group_widget.dart:219]: duplicated. please avoid it.

File: lib/presentation/widget/upload_request_group/created/created_upload_request_group_viewmodel.dart [hunk 5]
```diff
@@ -34,34 +34,25 @@ import 'dart:async';
...
-import 'package:linshare_flutter_app/presentation/view/context_menu/context_menu_builder.dart';
-import 'package:linshare_flutter_app/presentation/view/header/simple_bottom_sheet_header_builder.dart';
-import 'package:linshare_flutter_app/presentation/view/order_by/order_by_dialog_bottom_sheet.dart';
-import 'package:linshare_flutter_app/presentation/widget/base/base_viewmodel.dart';
+import 'package:linshare_flutter_app/presentation/widget/upload_request_group/upload_request_group_tab_common_viewmodel.dart';
 import 'package:redux/redux.dart';
 import 'package:redux_thunk/redux_thunk.dart';
 
-class CreatedUploadRequestGroupViewModel extends BaseViewModel {
+class CreatedUploadRequestGroupViewModel extends UploadRequestGroupTabViewModel {
```
Comment (hoangdat) [inline lib/presentation/widget/upload_request_group/created/created_upload_request_group_viewmodel.dart:46]: 👍

**Review by hoangdat — APPROVED** [review]

**Review by hoangdat — APPROVED** [review]

**Pull request merged**